hr_employee_calendar_planning/models/hr_employee.py

```diff
@@ -65,13 +63,23 @@ class HrEmployeeCalendar(models.Model):
          'Date end should be higher than date start'),
     ]
 
+    @api.model
     def create(self, vals):
         record = super(HrEmployeeCalendar, self).create(vals)
         record.employee_id._regenerate_calendar()
         return record
 
+    @api.multi
     def write(self, vals):
         res = super(HrEmployeeCalendar, self).write(vals)
         for employee in self.mapped('employee_id'):
             employee._regenerate_calendar()
         return res
+
+    @api.multi
+    def unlink(self):
+        employees = self.mapped('employee_id')
+        res = super(HrEmployeeCalendar, self).unlink()
+        for employee in employees:
+            employee._regenerate_calendar()
```

Comment:
This doesn't serve and can lead to errors. You only need to remove manually the linked specific calendar

Reply:
Without this unlink the user can delete `hr.employee.calendar` lines from an employee (in case he did some mistakes) and the resource calendar behind will not be updated, which in my opinion is not correct.

Reply:
Sorry, I have mixed models. I was talking about unlink in `hr.employee`, which should remove the linked private calendar. Yes, this is OK.

Reply:
Another case where it is not regenerated is when there are changes in attendance lines of a calendar. Again, I know these changes should only be done when the user made a mistake configuring the calendar, but do you think we should take them into account just in case?

Reply:
In this case I see a possible infinite loop problem and a lot of code for controlling that, so I prefer to not put it. You can put a button next to the calendar mapping table to manually regenerate the calendar as a "workaround".

Reply:
Putting a button could lead to confusion, the user may not now what 'Regenerate' means or can be hard to know which users you have to regenerate and when. 
I tried doing what I told you before and the infinite loop could happen theorically if the resource calendar of an employee has `active` as True. In practice, this is not possible since the only ways of modifiying the resource calendar of an employee are with the hook, which sets active to False, and regenerate, which also sets it to False.

Reply:
OK, try it then

Reply:
After trying it a lot I haven't had any problem with the infinite loop, so I think we can keep this feature to make sure that the calendar planning and the resource_calendar are always consistent.